python/raydp/torch/estimator.py

```diff
@@ -16,25 +16,98 @@
 #
 
 import inspect
-from typing import Any, Callable, List, NoReturn, Optional, Union
+from typing import Any, Callable, List, NoReturn, Optional, Union, Dict
+from numpy import outer
 
 import torch
-from ray.util.data import MLDataset
-from ray.util.sgd.torch.torch_dataset import TorchMLDataset
-from ray.util.sgd.torch.torch_trainer import TorchTrainer
-from ray.util.sgd.torch.training_operator import TrainingOperator
 from torch.nn.modules.loss import _Loss as TLoss
-from torch.utils.data.dataloader import DataLoader
 
 from raydp.estimator import EstimatorInterface
-from raydp.spark import RayMLDataset
 from raydp.spark.interfaces import SparkEstimatorInterface, DF, OPTIONAL_DF
 
+from ray import train
+from ray.train import Trainer, TrainingCallback, get_dataset_shard
+import ray.data
+from ray.data.dataset import Dataset
+from ray.data.impl.arrow_block import ArrowRow
+
+def train_fun(config):
```

Comment:
Is this configurable now? Maybe we should make this function configurable for some advanced use, what do you think?

Reply:
The behavior is same as before, so it's ok. I just thought it would be good if users are able to provide their train_fun, when the default one is not enough. For example, if they want to train a GAN network, then this is not enough, right?